pkg/vgmanager/vgmanager_controller.go

```diff
@@ -111,64 +157,72 @@ func (r *VGReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
 		deviceClassMap[deviceClass.Name] = i
 	}
 
-	// filter out block devices
-	remainingValidDevices, delayedDevices, err := r.filterAvailableDevices(blockDevices)
-	if err != nil {
-		_ = err
-	}
+	_, found := deviceClassMap[volumeGroup.Name]
 
-	for _, deviceClass := range lvmCluster.Spec.DeviceClasses {
-		// ignore deviceClasses whose LabelSelector doesn't match this node
-		// NodeSelectorTerms.MatchExpressions are ORed
-		node := &corev1.Node{}
-		selectsNode, err := NodeSelectorMatchesNodeLabels(node, deviceClass.NodeSelector)
-		if err != nil {
-			r.Log.Error(err, "failed to match nodeSelector to node labels")
-			continue
-		}
-		if !(selectsNode && ToleratesTaints(deviceClass.Tolerations, node.Spec.Taints)) {
-			continue
+	if len(matchingDevices) == 0 {
+		r.Log.Info("no matching devices", "VGName", volumeGroup.Name)
+		if len(delayedDevices) > 0 {
+			return reconcileAgain, nil
+		} else {
+			if found {
+				// Update the status again just to be safe.
+				if statuserr := r.updateStatus(ctx, status, volumeGroup); statuserr != nil {
```

Comment:
Yes, because the lvmd.yaml file is only updated once the VG is created. This block is mainly to handle cases when the status might not have been updated correctly. That function can also be modified to not update if there is no change for this entry.